src/lib/tests/symlink-manage.test.ts

```diff
@@ -0,0 +1,195 @@
+
+			// Mock fs.lstat to indicate these are symlinks
+			( fs.lstat as jest.Mock ).mockResolvedValue( { isSymbolicLink: () => true } as Stats );
+
+			// Mock fs.realpath to return a target for each symlink
+			( fs.realpath as jest.Mock ).mockImplementation( ( p: string ) =>
+				Promise.resolve( `/real/path/to/${ path.basename( p ) }` )
+			);
+
+			await symlinkManager.scanAndCreateSymlinks();
+
+			// Verify that PHP mkdir and mount were called for each symlink
+			expect( mockPHP.mkdir ).toHaveBeenCalledTimes( 2 );
+			expect( mockPHP.mount ).toHaveBeenCalledTimes( 2 );
+
+			// Verify that the symlinks were added to the internal map
+			expect( ( symlinkManager as unknown as SymlinkManagerPrivateProperties ).symlinks.size ).toBe(
+				2
+			);
+		} );
+	} );
+
+	describe( 'startWatching and stopWatching', () => {
+		it( 'should start and stop watching for file changes', async () => {
+			const mockWatcher = {
+				[ Symbol.asyncIterator ]: jest.fn().mockImplementation( () => ( {
+					next: jest.fn().mockResolvedValue( { done: true, value: undefined } ),
+				} ) ),
+			};
+			jest.spyOn( fs, 'watch' ).mockReturnValue( mockWatcher as never );
+
+			const watchPromise = symlinkManager.startWatching();
+			expect( fs.watch ).toHaveBeenCalledWith( mockProjectPath, expect.any( Object ) );
+
+			await symlinkManager.stopWatching();
+			await watchPromise;
+
+			expect( ( symlinkManager as unknown as SymlinkManagerPrivateProperties ).watcher ).toBeNull();
+			expect(
+				( symlinkManager as unknown as SymlinkManagerPrivateProperties ).watcherAbortController
+			).toBeNull();
+		} );
+	} );
+
+	describe( 'file change handling', () => {
+		it( 'should handle symlink creation', async () => {
+			mockPHP.fileExists.mockImplementation( ( path ) => ! path.includes( 'vfspath' ) );
+			mockPHP.readlink.mockImplementation( ( path ) => '/mock/document/root/vfspath/' + path );
+			( pathExists as jest.Mock ).mockResolvedValue( true );
+			const mockWatcher = {
+				[ Symbol.asyncIterator ]: jest.fn().mockImplementation( () => ( {
+					next: jest
+						.fn()
+						.mockResolvedValueOnce( {
+							done: false,
+							value: { eventType: 'rename', filename: 'newSymlink' },
+						} )
+						.mockResolvedValueOnce( { done: true, value: undefined } ),
+				} ) ),
+			};
+			jest.spyOn( fs, 'watch' ).mockReturnValue( mockWatcher as never );
+
+			( fs.lstat as jest.Mock ).mockResolvedValue( { isSymbolicLink: () => true } as Stats );
+			( fs.realpath as jest.Mock ).mockResolvedValue( '/real/path/to/newSymlink' );
+
+			const watchPromise = symlinkManager.startWatching();
+
+			await new Promise( ( resolve ) => setTimeout( resolve, 100 ) );
+
+			await symlinkManager.stopWatching();
+			await watchPromise;
+
+			expect( mockPHP.mkdir ).toHaveBeenCalled();
+			expect( mockPHP.mount ).toHaveBeenCalled();
+			const privateProps = symlinkManager as unknown as SymlinkManagerPrivateProperties;
+			expect( privateProps.symlinks.has( 'newSymlink' ) ).toBe( true );
```

Comment:
The React testing library provides a way to deal with asynchronous code via `waitFor`. Here's a potential update on on the code:

```suggestion
			await symlinkManager.startWatching();
			await waitFor( () => {
				expect( mockPHP.mkdir ).toHaveBeenCalled();
				expect( mockPHP.mount ).toHaveBeenCalled();
				const privateProps = symlinkManager as unknown as SymlinkManagerPrivateProperties;
				expect( privateProps.symlinks.has( 'newSymlink' ) ).toBe( true );
			} );
			await symlinkManager.stopWatching();
```

Reply:
Forgot to mention, but we need to import `waitFor`:
```
import { waitFor } from '@testing-library/react';
```